src/log.c

```diff
@@ -652,47 +652,26 @@ raft_entry_t *RaftLogGet(RaftLog *log, raft_index_t idx)
 
 RRStatus RaftLogDelete(RaftLog *log, raft_index_t from_idx)
 {
-    off_t offset;
-    RRStatus ret = RR_OK;
-
     if (from_idx <= log->snapshot_last_idx) {
         return RR_ERROR;
     }
 
-    while (log->index >= from_idx) {
-        if (!(offset = seekEntry(log, log->index))) {
+    raft_index_t count = log->index - from_idx + 1;
+    if (count != 0) {
+        off_t entry_offset = seekEntry(log, from_idx);
+        if (entry_offset == 0) {
             return RR_ERROR;
         }
 
-        RawLogEntry *re;
-        raft_entry_t *e;
-
-        if (readRawLogEntry(log->file, &re) < 0) {
-            ret = RR_ERROR;
-            break;
+        if (ftruncate(fileno(log->file), entry_offset) < 0) {
```

Comment:
Did we have in the past different types of log entries? what was the reason for `ENTRY` comparison?  what was the reason for the entry validation here?

Reply:
I guess there was some sort of an experimentation with different types of entries. Not sure what was the experiment though, can't relate with Raft.

Reply:
k, so we can treat it as leftover probably...